File: app/src/main/java/app/deadmc/devnetworktool/modules/PingStats.java
package app.deadmc.devnetworktool.modules;

import java.math.RoundingMode;
import java.text.DecimalFormat;

/**
 * Created by Feren on 30.11.2016.
 */
public class PingStats {

    private int sended;
    private int successful;
    private int failed;
    private int ttl;

    private float max;
    private float min;
    private float average;

    private float all;

    private String url;
    private String ipAddress;

    public PingStats() {
        sended = 0;
        successful = 0;
        failed = 0;
        ttl = 0;
        max = 0f;
        min = 0f;
        average = 0f;
        all = 0f;
        ipAddress = "";
    }

    public void setIpAddress(String ipAddress) {
        this.ipAddress = ipAddress;
    }

    public void setTtl(int ttl) {
        this.ttl = ttl;
    }

    public void addPing(float time) {
        sended++;
        if (time > 0.1f) {
            successful++;

            if (min == 0f)
                min = time;

            if (time < min)
                min = time;
            if (time > max)
                max = time;
            all+=time;
            average = all/sended;
        } else {
            failed++;
        }
    }

    public int getSended() {
        return sended;
    }

    public int getSuccessful() {
        return successful;
    }

    public int getFailed() {
        return failed;
    }

    public int getTtl() {
        return ttl;
    }

    public float getMax() {
        return max;
    }

    public float getMin() {
        return min;
    }

    public float getAverage() {
        return Math.round(average*10)/10f;
    }

    public String getIpAddress() {
        return ipAddress;
    }

}

File: app/src/main/java/app/deadmc/devnetworktool/models/PingStats.kt
package app.deadmc.devnetworktool.models

class PingStats {

    var sended: Int = 0
        private set
    var successful: Int = 0
        private set
    var failed: Int = 0
        private set
    var ttl: Int = 0

    var max: Float = 0.toFloat()
        private set
    var min: Float = 0.toFloat()
        private set
    var average: Float = 0.toFloat()
        get() = Math.round(field * 10) / 10f

    private var all: Float = 0.toFloat()

    private val url: String? = null
    var ipAddress: String? = null

    init {
        sended = 0
        successful = 0
        failed = 0
        ttl = 0
        max = 0f
        min = 0f
        average = 0f
        all = 0f
        ipAddress = ""
    }

    fun addPing(time: Float) {
        sended++
        if (time > 0.1f) {
            successful++

            if (min == 0f)
                min = time

            if (time < min)
                min = time
            if (time > max)
                max = time
            all += time
            average = all / sended
        } else {
            failed++
        }
    }

}
